embedded-graphics/MIGRATING-0.5-0.6.md

```diff
@@ -0,0 +1,248 @@
+# Migrating from embedded-graphics 0.5.x to 0.6.0
+
+## Pixel colours
+
+TODO for both driver authors and library consumers
```

Comment:
@rfuest As you did most of the colour changes, do you have any suggestions for what to add here? A bulleted list is fine - I'd just like a starting point to work from.

Reply:
Some initial suggestions:

* List of color types.
* Primitive types (like `u8` or `u16`) can't directly be used as colors anymore.
  The primitive types didn't allow to specify the exact color format on a type level, which could lead to errors when colors weren't used in different places without being converted explicitly (e.g. when displaying an image).
* Added distinction between the pixel color type and the underlying storage.
  Multiple color types can now share the same storage size, but still have distinct types.
* Added some color conversions:
  * Any RgbColor <-> Any other RgbColor
  * Any GrayColor -> Any RgbColor
  * BinaryColor -> Any RgbColor or GrayColor

For display driver authors:
* Use `BinaryColor` when implementing a driver for a display with 2 states.
* When implementing a color display driver `DrawTarget` can be implemented to automatically convert colors:
  ```rust
  impl<C> DrawTarget<C> for SomeRgb565Display
  where
    C: PixelColor + Into<Rgb565>,
  {
       type Error = core::convert::Infallible;
  
      fn draw_pixel(&mut self, pixel: Pixel<C>) -> Result<(), Self::Error> {
          let Pixel(point, color) = pixel;
  
          self.set_pixel(point.x, point.y, color.into());
  
          Ok(())
      }
  
      fn size(&self) -> Size {
          self.size
      }
  }
  ```
  This way, for example, a `Rgb565` or `Rgb888` image can be drawn the same way, without any explicit color conversion.